packages/vite/src/node/plugins/asset.ts

```diff
@@ -182,6 +185,12 @@ export function assetPlugin(config: ResolvedConfig): Plugin {
     },
 
     generateBundle(_, bundle) {
+      duplicateAssets.forEach((asset) => {
+        if (asset.name) {
```

Comment:
Why is this guard needed? `name` should always be defined here, no?

Reply:
`OutputAsset.name` is `string|undefined` so TS was screaming at me. Should I remove it as we know at runtime it is set?

Reply:
Yes, lets use `!` and a comment here. If not it looks like some duplicated assets may be dropped

Reply:
Fixed! Thank you